### ISSUE TITLE
Consider providing MockMvcHtmlUnitDriverBuilder.build() instead of .createDriver()

### ISSUE THREAD

**Issue by chrylis** [spring-projects/spring-test-htmlunit#54]
The builder method on `MockMvcHtmlUnitDriverBuilder` is named `createDriver()`. Is there a particular reason for that name and not `build()`?

**Comment by rwinch** [comment]
I thought about this, but I find the name build too generic. Did you have a reason you are asking?

**Comment by chrylis** [comment]
It was just surprising. It seems that all the other builders everywhere use `build()`, and I was surprised to get "dynamic method" highlighting on it.

**Comment by rwinch** [comment]
Thanks for the response.

> It seems that all the other builders everywhere use build()

I think build() will hide the option for configureDriver which is necessary (as described earlier). Perhaps we should consider having an alias. I have updated this issue to be considered vs being a question, but it depends on how much feedback we get.

> I was surprised to get "dynamic method" highlighting on it.

I assume you mean when using Groovy? If so, perhaps it is due to generics and you invoked a method on the base class first?

For example if you have something like this I have seen IDEs treat it as dynamic method due to poor handling of the generic types

```groovy
MockMvcHtmlUnitDriverBuilder.webAppContextSetup(context)
    .contextPath("") // returns a generic that extends base class
    .createDriver()
```

If you use Java it works, so any dynamic method highlighting would be an IDE/groovy issue.

**Comment by chrylis** [comment]
> I assume you mean when using Groovy? If so, perhaps it is due to generics and you invoked a method on the base class first?

No, I simply mean that I assumed the method name would be `build()`, and then Eclipse told me that the method didn't exist. `createDriver()` is resolved with no problem.

**Comment by rwinch** [comment]
Ahh...that makes sense. Thank you for clarifying.

**Comment by rwinch** [comment]
This project is being merged into spring-test-4.2.0.RELEASE. We have renamed the method to .build() in the code that has been merged.

Thanks again for the report.
